lib/improver/blending/weighted_blend.py

```diff
@@ -573,12 +568,17 @@ def shape_weights(self, cube, weights):
             # Multi-dimensional weights array provided that matches cube.
             weights_array = weights.data.astype(np.float32)
         else:
-            # 1D array of weights across the blending coordinate.
-            coord_dim_thres = cube.coord_dims(self.coord)
+            # Map array of weights to shape of cube to collapse.
```

Comment:
Before this function could only cope with weights with 1D or weights that exactly matched the dims on the cube to collapse. I've updated the broadcasting to be a bit more general, so one unit test now passes when it failed (we need to add some expected data to this test), but we probably need a few more unit tests to test this broadcasting properly.
See details on the ticket.